experimental/plugin.go

```diff
@@ -0,0 +1,30 @@
+package experimental
+
+import (
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+// PluginExe is the singleton container for your plugin.
+type PluginExe interface {
+	// CheckHostHealth for the plugin executable
+	CheckExeHealth(config backend.PluginConfig) *backend.CheckHealthResult
+
+	// request for a new datasource
+	NewDataSourceInstance(config backend.PluginConfig) (DataSourceInstance, error)
+}
+
+// DataSourceInstance implements each of the supported requests.  Alternativly this could be a set
+// of interfaces that work for DataSource|AlertNotifier|etc|etc... with the support interrogated and
+// maintaind by the helper on startup
+type DataSourceInstance interface {
+	CheckHealth() *backend.CheckHealthResult
+
+	// If the request does not need access to the headers or user, use this request
```

Comment:
The documentation comment should be formed as a sentence, starting with QueryData.